### DISCUSSION TITLE
help with set up

### DISCUSSION THREAD

**Discussion by yaowzhang** [saleweaver/python-amazon-sp-api#796] [FAQ]
I am not sure how to even get this started.  I have tried for hours to get the code working but it constantly returning error, my Python level is limited which did not help.  How can I get some hand holding to get myself set up with this lib so that I can start play with it.  I apologize if this is not appropriate.

**Comment by osvelqbano** [comment]
Let me try to help out, they say the best way to learn is to teach so here goes nothing. FYI I'm not an expert but Im able to make calls using this library 

1. Did you do all of the APP Set Up in Seller Central and AWS ? Can you fill all this out 
credentials = dict(
    refresh_token='**your_refresh_token**',  # From Seller central under Authorise -> Refresh Token
    lwa_app_id='**your_lwa_app_id**',  # From Seller Central, named CLIENT IDENTIFIER on website.
    lwa_client_secret='**your_lwa_client_secret**',  # From Seller Central, named CLIENT SECRET on website.
    aws_access_key='**your_aws_access_key**',  # From AWS IAM Setup
    aws_secret_key='**your_aws_secret_key**',  # From AWS IAM Setup
    role_arn='**your_role_arn**'  #arn:aws:iam::1234567890:role/SellingPartnerAPIRole
)
If not you need  to do these tutorials
[Tutorial : 1](https://developer-docs.amazon.com/sp-api/docs/creating-and-configuring-iam-policies-and-entities)
[Tutorial : 2 ](https://developer-docs.amazon.com/sp-api/docs/self-authorization)

Then Install the Library 
Pip Install : pip install python-amazon-sp-api

Import the endpoints that you will use 
from sp_api.api import (endpoint Name)  -- Example: from sp_api.api import Orders

then call what you imported in the following manner 

test = Orders(credentials = credentials ) -- Credentials are the items on 1

then use the classes 

test_orders = test.**class()**   -- replace class() with actual classes such as get_order() enter the keyword inside the parenthesis 

test_order = test.get_order('Your Order Number') 

rinse and repeat for each endpoint
